cli/cook/mesos.py

```diff
@@ -0,0 +1,39 @@
+import logging
+from urllib.parse import urlparse
+
+from cook import http
+
+
+def instance_to_agent_url(instance):
+    """
+    Given a job instance, returns the base Mesos agent URL,
+    e.g. http://agent123.example.com:5051
+    """
+    hostname = instance['hostname']
+    if 'output_url' in instance:
+        output_url = instance['output_url']
+        url = urlparse(output_url)
+        netloc = url.netloc
+    else:
+        logging.info('assuming default agent port of 5051')
+        netloc = f'{hostname}:5051'
+    return f'http://{netloc}'
+
+
+def retrieve_instance_sandbox_directory(instance, job):
```

Comment:
If the output url exists for the job, you can get the sandbox directory from that with some shitty parsing. 

You should also check if the sandbox dir is set for the instance and just use that if it is.

Reply:
> If the output url exists for the job, you can get the sandbox directory from that with some shitty parsing.

Good call.

> You should also check if the sandbox dir is set for the instance and just use that if it is.

What do you meany by "check if the sandbox dir is set for the instance"?

Reply:
:+1: I see now that the instance can contain a `sandbox_directory` field directly. Checking for that now too.